shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.m

```diff
@@ -254,4 +256,49 @@ - (void)testUITextInputCallsUpdateEditingStateOnce {
   [inputView unmarkText];
   XCTAssertEqual(updateCount, 6);
 }
+
+- (void)testUpdateFirstRectForRange {
+  FlutterTextInputView* inputView = [[FlutterTextInputView alloc] init];
+  [inputView
+      setTextInputState:@{@"text" : @"COMPOSING", @"composingBase" : @1, @"composingExtent" : @3}];
+
+  CGRect kInvalidFirstRect = CGRectMake(-1, -1, 9999, 9999);
+  FlutterTextRange* range = [FlutterTextRange rangeWithNSRange:NSMakeRange(0, 1)];
+  // yOffset = 200.
+  NSArray* yOffsetMatrix = @[ @1, @0, @0, @0, @0, @1, @0, @0, @0, @0, @1, @0, @0, @200, @0, @1 ];
+  NSArray* zeroMatrix = @[ @0, @0, @0, @0, @0, @0, @0, @0, @0, @0, @0, @0, @0, @0, @0, @0 ];
```

Comment:
Nit: Would it be more clear if you formatted these arrays like a matrix?  Like indent them to be a 4x4 grid I mean.  Up to you if that is an improvement or not.

Reply:
It was originally formatted like that. Then the formatter made them one line.